tower_cli/resources/job_template.py

```diff
@@ -145,3 +146,18 @@ def disassociate_notification_template(self, job_template,
         """Disassociate a notification template from this job template."""
         return self._disassoc('notification_templates_%s' % status,
                               job_template, notification_template)
+
+    @resources.command(use_fields_as_options=('extra_vars'))
+    @click.option('--host-config-key', help='Job-template-specific string used to authenticate '
+                  'host during provisioning callback.')
+    def callback(self, pk=None, host_config_key='', extra_vars=None):
+        """Contact Tower and request a configuration update using this job template."""
+        url = self.endpoint + '%s/callback/' % pk
+        if not host_config_key:
+            host_config_key = client.get(url).json()['host_config_key']
```

Comment:
It is true that provisioning callback does not need require normal Tower authentication at all. The latest patch solves the issue by explicitly set `auth=None`.